site-src/geps/gep-1058.md

```diff
@@ -0,0 +1,623 @@
+Configurations](#configuration-conflicts-and-invalid-configurations) section,
+above.
+
+### Example Reconciliation of Parent and Child Routes
+
+The diagram below shows how a parent route and two child route are reconciled
+(i.e. combined) and the ephemeral route configuration that is the result. These
+route configurations match the Scenario 1 diagram shown previously. To keep this
+diagram from being even larger than it is, the routing configuration for the
+`api` namespace is not included.
+
+![Reconciling Routes Example](images/1058-reconciling-routes.png "Example of Reconciling Routes")
+[Click here to see diagram in full size ](images/1058-reconciling-routes.png)
+
+### Summary of Changes, Requirements and Constraints: Binding and Reconciling Routes
+
+1. The process of binding a child route to a parent route is treated as a single
+   transaction
+   1. Any non-recoverable error encountered in the process of reconciling and
+      binding the two routes, is treated as fatal to the transaction and the
+      entire transaction fails.
+   2. A failed attempt to bind a child route to a parent route shall result in
+      no changes to the configuration of the active Ephemeral route.
+2. A failed binding between and parent and child routes shall not prevent either
+   route from binding with other routes.
+3. When multiple child routes are configured to bind to the same `RouteRule` in
+   a parent route, the order in which the child routes bind to the parent should
+   be assumed to be random. They are not guarantied to be in any specific order.
+   1. Implementations may allow other combinations of path matching types. If
+      they do, it should be clearly documented which combinations are allowed
+      and how the values in each combination are reconciled.
+4. When reconciling a parent route and a child route (each being individually
+   valid), if a combination of settings in the two routes would result in an
+   Ephemeral route configuration that would not be valid as a standalone route,
+   that is to be considered a config conflict between the two routes and shall
+   be treated as a non-recoverable error. This shall result in the bind
+   transaction failing.
+5. When reconciling a parent route and a child route, The `RouteMatch`\(s\) and
+   `RouteFilter`\(s\) from the Parent's `RouteRule` are AND'ed those in each
+   `RouteRule` in the child route. This creates a set of `RouteRule`\(s\) that
+   are added to the Ephemeral route.
+   1. If the child route attaches to multiple `RouteRule`\(s\) in the parent,
+      each `RouteRule` from the parent is independently reconciled with the
+      child's `RouteRules`\(s\). For each `RouteRul` in the parent, an
+      additional set of `RouteRule`\(s\) is created and added to the Ephemeral
+      route.
+6. Path Matching: If `HTTPPathMatch` is configured in both the parent and child
+   `Routes`, allow **only** substring prefix matching on path, without any
+   wildcard or RegEx matching,
```

Comment:
I think path matching will always be configured because our default path match is prefix + "/". A lot of these limitations seem like things we could enforce with webhook validation. Maybe we should have a section covering what additions we'll need to webhook validation.